src/functions.py

```diff
@@ -517,6 +517,8 @@ def crossoverSinglePoint(parent1, parent2):
             flatParam1 = np.ravel(param1)
             flatParam2 = np.ravel(param2)
 
+            randIndex = int(random.random()*flatParam1.size)
```

Comment:
Good fix ⬆️ 

randIndex was set to 1 in the code commited to main, which is why it was removed/cleaned up.